### DISCUSSION TITLE
Automation - create row with relationships

### DISCUSSION THREAD

**Discussion by andyburgessmd** [Budibase/budibase#9894] [Help] [answered]
Hoping someone can help with an issues I am having with automation.
The automation is triggered by a row creation (table a) and it then needs to create a row in another table (table b) using some data from the triggered row (table a) and some of the fields/columns in table b are relationships.
What is the format of the data that should be set in the create row fields for table b where it is relational? Is it just the _id or does it require more?

Example if I want to use the 'Created By' user from the trigger.row do I use just the _id from trigger.row."Created By" or the whole "Created By" object? Or do I need to make an object like {"_id":"ro_ta_xxxxxx"} with the _id value?

**Comment by andyburgessmd** [comment]
**Extra info**
- This is using a budibasedb table
- The automation is being tested in a published app 
- I've tried running all sorts of combinations but not had any success
- The automations don't run or show in the automations history when they fail

**Comment by melohagan** [comment]
Hey @andyburgessmd 

Would you be able to provide me a little more information about tables A and B. Using actual names would help me visualise this, and would you be able to share the fields involved (names and types) and some screenshots of the automation that you have so far please?

**Comment by andyburgessmd** [comment]
Sure. 
**Table A - Requests**
Lots of fields but has one is called Approvals which is a relationship type to Approvals table
![image](https://user-images.githubusercontent.com/110813989/223437361-2fcb2ab0-4fb4-46bf-af31-1ee326ac5ef0.png)
And another is an 'Owner' which is a realtionship to the users table
![image](https://user-images.githubusercontent.com/110813989/223438804-b570dd54-6bfe-44a8-ae90-3a95dfe3621e.png)

**Table B - Approvals (many approvals -> one request)**
![image](https://user-images.githubusercontent.com/110813989/223437646-b9b6cb5c-19e8-4559-8362-2849b43d7fcc.png)

The automation I am trying to run is that when a new row is created in the Requests table the selected Owner is added into the Approvals table with the id of request row added and a value of Pending in the status. 

Current automation setup is 
![image](https://user-images.githubusercontent.com/110813989/223439525-95ed6886-e5d2-445e-b202-3ec349ce9347.png)
using 
`return $("trigger.row.._id");`
and 
`return $("trigger.row.Owner.0._id");`
as the values for Requests and User.

I've tried using the handlebars and JS bindings and have output them to backend logs to check the values.

**Comment by mctlisboa** [comment, marked as answer]
The previous step is returning the user id row as an object like "ro_ta_users_us_76746f125e3d4232bf2515f42d02b89b"

But Budibase automation to sucessfully map relationships probably requires a string. So by wrapping inside a ```{{ JSONstringify }}``` I was able to run the automation successfully.

**Reply by melohagan** [reply]
Also a fix out here: https://github.com/Budibase/budibase/pull/9924
